snc.sh

```diff
@@ -35,6 +36,66 @@ else
     fi
 fi
 
+function preflight_failure() {
+        local msg=$1
+        echo "$msg"
+        if [ -z "${SNC_NON_FATAL_PREFLIGHT_CHECKS-}" ]; then
+                exit 1
+        fi
+}
+
+function run_preflight_checks() {
+        echo "Checking libvirt and DNS configuration"
+
+        LIBVIRT_URI=qemu+tcp://localhost/system
+
+        # check if libvirtd is listening on a TCP socket
+        if ! virsh -c ${LIBVIRT_URI} uri >/dev/null; then
+                preflight_failure  "libvirtd is not listening for plain-text TCP connections, see https://github.com/openshift/installer/tree/master/docs/dev/libvirt#configure-libvirt-to-accept-tcp-connections"
+        fi
+
+        # check if libvirtd has access to an x86_64 hypervisor
+        local arch
+        arch=$(virsh  capabilities | xmlstarlet sel -t -v '/capabilities/host/cpu/arch')
+        if [ ${arch} != "x86_64" ]; then
+                # warn only
+                echo "The host architecture is ${arch}, SNC has only been tested on x86_64"
+        fi
+
+        # check for availability of a hypervisor using kvm
+        if ! virsh -c ${LIBVIRT_URI} capabilities | ${XMLSTARLET} sel -t -c "/capabilities/guest/arch[@name=\"${arch}\"]/domain[@type=\"kvm\"]" >/dev/null; then
+                preflight_failure "Your ${arch} platform does not provide a hardware-accelerated hypervisor, it's strongly recommended to enable it before running SNC. Check virt-host-validate for more detailed diagnostics"
+                return
+        fi
+
+        # check that api.crc.testing either can't be resolved, or resolves to 192.168.126.1[01]
+        local ping_status
+        ping_status="$(ping -c1 api.crc.testing | head -1 || true >/dev/null)"
+        if echo ${ping_status} | grep "PING api.crc.testing (" && ! echo ${ping_status} | grep "192.168.126.1[01])"; then
+                preflight_failure "DNS setup seems wrong, api.crc.testing resolved to an IP which is neither 192.168.126.10 nor 192.168.126.11, please check your NetworkManager configuration and /etc/hosts content"
+                return
+        fi
+
+        # check if firewalld is configured to allow traffic from 192.168.126.0/24 to 192.168.122.1
+        # this check is very basic and expects the configuration to match
+        # https://github.com/openshift/installer/tree/master/docs/dev/libvirt#firewalld
+        local zone
+        if firewall-cmd -h >/dev/null; then
+                # With older libvirt, the 'libvirt' zone will not exist
+                if firewall-cmd --get-zones |grep '\<libvirt\>'; then
+                        zone=libvirt
+                else
+                        zone=dmz
+                fi
+                if ! firewall-cmd --zone=${zone} --list-services | grep '\<libvirt\>'; then
```

Comment:
This does not work so well on a stock rhel8 install, fails with
```
Authorization failed.
    Make sure polkit agent is running or run the application as superuser.
```
Maybe best to skip this check for now.

Reply:
@cfergeau sure, let's update the PR with the changes.